frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
+                    className={`font-mono-sm border-1px bg-primary-lighter radius-md padding-left-1 padding-right-1 margin-top-0 ${
+                        uppercaseMain === StatusEnum.BATCH && "bg-blue-5"
+                    } ${uppercaseMain === StatusEnum.PROCESS && "bg-blue-10"} ${
+                        uppercaseMain === StatusEnum.READY && "bg-blue-20"
+                    }`}
+                >
+                    {uppercaseMain}
+                </p>
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Sub)}>
+                {receiver.Sub}
+            </td>
+            <td
+                className={`message-receiver-break-word white-space-pre ${checkForActiveColumn(
+                    ColumnDataEnum.FileName
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.FileName}:`,
+                        body: receiver.FileName,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.FileName}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.TransportResults}:`,
+                        body: receiver.TransportResults,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.TransportResults}
+            </td>
+        </tr>
+    );
 };
 
 export const MessageReceivers = ({ receiverDetails }: MessageReceiverProps) => {
+    const modalRef = useRef<ModalRef>(null);
+    const [modalText, setModalText] = useState({ title: "", body: "" });
+    const [activeColumn, setActiveColumn] = useState("");
+    const [activeColumnSortOrder, setActiveColumnSortOrder] = useState("");
+    const filterIcon = useMemo(
+        () =>
+            activeColumnSortOrder === FilterOptionsEnum.ASC ? (
+                <Icon.ArrowUpward />
+            ) : activeColumnSortOrder === FilterOptionsEnum.DESC ? (
+                <Icon.ArrowDownward />
+            ) : (
+                <Icon.SortArrow />
```

Comment:
Also, minor point, but would you mind unnesting this ternary?  I think an `if-else if-else` or even a `case` could be clearer. 

We could enable [an ESLint rule](https://eslint.org/docs/latest/rules/no-nested-ternary) for preventing nested ternaries too 🤷

Reply:
Done with `filterIcon` refactor!